### ISSUE TITLE
[Vault KMS] don't assume the `transit` path

### ISSUE THREAD

**Issue by raffaelespazzoli** [sigstore/cosign#2651]
**Description**

currently cosing assumes that the transit secret engine will be mounted at the `transit` path all of the times.
This is not the case for Vault. The mount path should be explicit in the kms URL.

**Comment by hasanhakkaev** [comment]
Hi,
I guess can find the solution here > https://docs.sigstore.dev/cosign/kms_support/#hashicorp-vault

**Comment by hectorj2f** [comment]
@raffaelespazzoli Indeed, you can find how to point to the transit path, as @hasanhakkaev mentioned above.

**Comment by hasanhakkaev** [comment]
Not much of a news for this particular case but, [here](https://github.com/sigstore/sigstore/blob/main/pkg/signature/kms/hashivault/client.go#L137) is the "transit" secret engine hardocded
